src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key of Object.keys(action.extraData)) {
+          state[key] = action.extraData[key];
+        };
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+      case 'clear':
+        for (const key in state) {
```

Comment:
good job but there is some mistake
1.[ CODE STYLE]: don't use for in loop for iterating over array